demos/build_demos_msvc.bat

```diff
@@ -75,84 +77,54 @@ if "%PROCESSOR_ARCHITECTURE%" == "AMD64" (
     set "PLATFORM=Win32"
 )
 
-set VSWHERE="false"
-if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" (
-    set VSWHERE="true"
-    cd "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer"
-) else if exist "%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe" (
-    set VSWHERE="true"
-    cd "%ProgramFiles%\Microsoft Visual Studio\Installer"
-) else (
-    echo "vswhere tool is not found"
-)
-
-if !VSWHERE! == "true" (
-    if "!VS_VERSION!"=="" (
-        echo Searching the latest Visual Studio...
-        for /f "usebackq tokens=*" %%i in (`vswhere -latest -products * -requires Microsoft.Component.MSBuild -property installationPath`) do (
-            set VS_PATH=%%i
-        )
+if "%VS_VERSION%" == "" (
+    if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" (
+        set "VSWHERE=%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+    ) else if exist "%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe" (
+        set "VSWHERE=%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
     ) else (
-        echo Searching Visual Studio !VS_VERSION!...
-        for /f "usebackq tokens=*" %%i in (`vswhere -products * -requires Microsoft.Component.MSBuild -property installationPath`) do (
-            set CUR_VS=%%i
-            if not "!CUR_VS:%VS_VERSION%=!"=="!CUR_VS!" (
-                set VS_PATH=!CUR_VS!
-            )
-        )
-    )
-    if exist "!VS_PATH!\MSBuild\14.0\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\14.0\Bin\MSBuild.exe"
-    )
-    if exist "!VS_PATH!\MSBuild\15.0\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\15.0\Bin\MSBuild.exe"
-    )
-    if exist "!VS_PATH!\MSBuild\Current\Bin\MSBuild.exe" (
-        set "MSBUILD_BIN=!VS_PATH!\MSBuild\Current\Bin\MSBuild.exe"
+        echo vswhere.exe was not found.
+        echo Make sure Visual Studio 2017 version 15.2 or a newer version of Visual Studio
```

Comment:
You have 2017 here because older versions don't include vswhere, right? But this error message can be misleading because it says that Visual Studio 2017+ is required while it is till possible to pass 2015 as argument.

And if vswhere appeared in 2017 you don't need to handle `2015` value when you search `VS_VERSION` with vswhere below as there will be at least the version which contains vswhere.

Reply:
> You have 2017 here because older versions don't include vswhere, right? But this error message can be misleading because it says that Visual Studio 2017+ is required while it is till possible to pass 2015 as argument.

Okay, I've amended the message to make it more accurate.

> And if vswhere appeared in 2017 you don't need to handle 2015 value when you search VS_VERSION with vswhere below as there will be at least the version which contains vswhere.

vswhere appeared in 2017, but it _can_ report information about 2015, and I think that in rare cases 2015 can end up being reported as the latest (e.g. if both 2015 and 2017 are installed, but 2017 doesn't include the MSBuild component).

I should also note that from what I recall, VS2015 support is planned to be dropped in OpenVINO, so this whole point might be moot soon.

Reply:
Seems 2015 is dropped: https://github.com/openvinotoolkit/openvino/blob/master/build-instruction.md#software-requirements-1. But may be we could keep it as it costs only few ifs for us.